### PR TITLE
TST/CLN: parametrize and clean test_expressions, test_nanops

### PR DESCRIPTION
**Pull request by jbrockmendel**
These are each going to need at least one more pass after this.

### CODE REVIEW

File: pandas/tests/test_nanops.py [hunk 1]
```diff
@@ -152,14 +149,13 @@ def check_fun_data(
         targfunc,
         testarval,
         targarval,
-        targarnanval,
```
Comment (WillAyd) [inline pandas/tests/test_nanops.py]: Can you just comment on what this removal is?

File: pandas/tests/test_nanops.py [hunk 2]
```diff
@@ -355,22 +310,31 @@ def test_nanmean_overflow(self):
...
 
-    def test_returned_dtype(self):
-
-        dtypes = [np.int16, np.int32, np.int64, np.float32, np.float64]
-        if hasattr(np, "float128"):
-            dtypes.append(np.float128)
+    @pytest.mark.parametrize(
+        "dtype",
+        [
+            np.int16,
```
Comment (WillAyd) [inline pandas/tests/test_nanops.py]: Not here and no action required but maybe want to expand to cover unsigned and 8 bit versions. I've seen that in a few other PRs so should become a fixture if not already

File: pandas/tests/test_nanops.py [hunk 3]
```diff
@@ -355,22 +310,31 @@ def test_nanmean_overflow(self):
...
+            np.int32,
+            np.int64,
+            np.float32,
+            np.float64,
+            getattr(np, "float128", None),
```
Comment (WillAyd) [inline pandas/tests/test_nanops.py]: Is there a good way to convert this to a skip if not available?

File: pandas/tests/test_nanops.py [hunk 4]
```diff
@@ -300,33 +270,20 @@ def _badobj_wrap(self, value, func, allow_complex=True, **kwargs):
...
-            allow_date=False,
-            allow_tdelta=False,
+            nan_op, np_op, allow_all_nan=False, allow_date=False, allow_tdelta=False
         )
 
     def test_nansum(self):
         self.check_funs(
             nanops.nansum,
             np.sum,
-            allow_str=False,
```
Comment (WillAyd) [inline pandas/tests/test_nanops.py]: Is this just universally removed here?

**Review by WillAyd — COMMENTED** [review]
Thanks for paying this one right back ha...

Looks pretty good - definitely in need of this here. I don't think any actions required just some questions that may help to review

File: pandas/tests/test_nanops.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/test_nanops.py]: On 212-213 we check if `targarnan is None` and if so set `targarnan = testar`.  That condition _always_ holds in the test, so  ripped it out, and targarnanval as a consequence

File: pandas/tests/test_nanops.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/test_nanops.py]: Good idea

File: pandas/tests/test_nanops.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/test_nanops.py]: I guess we could make this:

```
pytest.param(getattr(np, "float128", None), marks=pytest.skipif(not hasattr(np, "float128"))
```

Tough call whether the explicit skip is worth the added verbosity

File: pandas/tests/test_nanops.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/tests/test_nanops.py]: yah, so just removed allow_str altogether.

File: pandas/tests/test_nanops.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/test_nanops.py]: we also have our fixtures for dtypes (not blocking this PR)

File: pandas/tests/test_nanops.py [hunk 5]
```diff
@@ -355,22 +310,31 @@ def test_nanmean_overflow(self):
...
-            group_a = ["mean", "std", "var", "skew", "kurt"]
-            group_b = ["min", "max"]
-            for method in group_a + group_b:
-                result = getattr(s, method)()
-                if is_integer_dtype(dtype) and method in group_a:
-                    assert result.dtype == np.float64
-                else:
-                    assert result.dtype == dtype
+        s = Series(range(10), dtype=dtype)
+        group_a = ["mean", "std", "var", "skew", "kurt"]
```
Comment (jreback) [inline pandas/tests/test_nanops.py]: likely can be parmaterize using our aggregation fixtures

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks, a couple of follow-on comments

**Pull request merged**